apps/src/netsim/NetSimEntity.js

```diff
@@ -110,3 +110,25 @@ NetSimEntity.prototype.getTable_ = function () {
 NetSimEntity.prototype.buildRow_ = function () {
   return {};
 };
+
+/**
+ * Destroys all provided entities (from remote storage) asynchronously, and
+ * calls onComplete when all entities have been destroyed and/or an error occurs.
+ * @param {NetSimEntity[]} entities
+ * @param {!NodeStyleCallback} onComplete
+ */
+NetSimEntity.destroyEntities = function (entities, onComplete) {
+  if (entities.length === 0) {
+    onComplete(null, true);
+    return;
+  }
+
+  entities[0].destroy(function (err, result) {
+    if (err) {
+      onComplete(err, result);
+      return;
+    }
+
+    NetSimEntity.destroyEntities(entities.slice(1), onComplete);
```

Comment:
It's like continuously feeding a caterpillar into a guillotine :bug: :hocho: :sob: